lib/posix/pthread.c

```diff
@@ -302,6 +309,35 @@ int pthread_getschedparam(pthread_t pthread, int *policy,
 	return 0;
 }
 
+/**
+ * @brief Dynamic package initialization
+ *
+ * See IEEE 1003.1
+ */
+int pthread_once(pthread_once_t *once, void (*initFunc)(void))
+{
+	if (atomic_get(once) == PTHREAD_ONCE_INIT) {
+		return 0;
+	}
+
+	if (atomic_cas(once, NOT_INITIALIZED, INITIALIZING)) {
```

Comment:
do only compare here and in the block increment.

Reply:
okay, implemented it this way